cmd/config_consolidation_test.go

```diff
@@ -375,6 +375,70 @@ func getConfigConsolidationTestCases() []configConsolidationTestCase {
 				assert.Equal(t, []string{"avg", "p(90)", "count"}, c.Options.SummaryTrendStats)
 			},
 		},
+		{opts{cli: []string{}}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.NewString("inf", false),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSFirst, Valid: false},
+			}, c.Options.DNS)
+		}},
+		{opts{env: []string{"K6_DNS=ttl=5,strategy=round-robin"}}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.StringFrom("5"),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSRoundRobin, Valid: true},
+			}, c.Options.DNS)
+		}},
+		{opts{env: []string{"K6_DNS=ttl=inf,strategy=random"}}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.StringFrom("inf"),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSRandom, Valid: true},
+			}, c.Options.DNS)
+		}},
+		{opts{cli: []string{"--dns", "ttl=-1"}}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.StringFrom("-1"),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSFirst, Valid: true},
+			}, c.Options.DNS)
+		}},
+		{opts{cli: []string{"--dns", "ttl=0"}}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.StringFrom("0"),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSFirst, Valid: true},
+			}, c.Options.DNS)
+		}},
+		{opts{cli: []string{"--dns", "ttl=5s,strategy="}}, exp{cliReadError: true}, nil},
+		{opts{fs: defaultConfig(`{"dns": {"ttl": "0", "strategy": "round-robin"}}`)}, exp{}, func(t *testing.T, c Config) {
+			assert.Equal(t, &lib.DNSConfig{
+				TTL:      null.StringFrom("0"),
+				Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSRoundRobin, Valid: true},
+			}, c.Options.DNS)
+		}},
+		{
+			opts{
+				fs:  defaultConfig(`{"dns": {"ttl": "0"}}`),
+				env: []string{"K6_DNS=ttl=30"},
+			},
+			exp{}, func(t *testing.T, c Config) {
+				assert.Equal(t, &lib.DNSConfig{
+					TTL:      null.StringFrom("30"),
+					Strategy: lib.NullDNSStrategy{DNSStrategy: lib.DNSFirst, Valid: true},
+				}, c.Options.DNS)
+			},
+		},
+		{
+			// FIXME: CLI arg *should* override all
```

Comment:
if you have a pointer, for `lib.Options.DNSConfig`, this is going to be nightmarishly difficult to fix with envconfig, so I suggest we just set it to `ignored:"true"` for now, i.e. unable to be configured by environment variables

Reply:
Hmm yeah, wouldn't having `NullDNSConfig` also solve of having `NullDNSStrategy`? A "higher" DNS setting will simply overwrite all "lower" DNS options with its own or the default.

Reply:
Yup, I abandoned the pointer approach... It was actually more nightmarish to deal with our insane `Apply()` mechanism than envconfig, but it seems to work now with a struct value, after a frustrating amount of trial and error. 🙈